File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).
-export([to_rna/1, dna_rna_map/1]).

dna_rna_map($G) -> $C;
dna_rna_map($C) -> $G;
dna_rna_map($T) -> $A;
dna_rna_map($A) -> $U.

to_rna(Strand) -> [dna_rna_map(Nucleotide) || Nucleotide <- Strand].

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  DNA_RNA_MAP = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }

  RNA_DNA_MAP = DNA_RNA_MAP.invert

  def self.find(nucleotides, map)
    nucleotides.
      chars.
      flat_map { |nucleotide| map.fetch(nucleotide) }.
      join
  rescue KeyError
    raise ArgumentError
  end

  def self.of_dna(nucleotides)
    find(nucleotides, DNA_RNA_MAP)
  end

  def self.of_rna(nucleotides)
    find(nucleotides, RNA_DNA_MAP)
  end
end
